### DISCUSSION TITLE
Always LOOK when you re-enter the room

### DISCUSSION THREAD

**Discussion by drrak** [okaybenji/text-engine#38] [Q&A] [answered]
_Translator:_
Great engine. I like everything very much, except for the abbreviated description after re-entering a previously visited room. Because of this, I often have to enter LOOK.

Can this be easily changed?

**Comment by dthigpen** [comment]
If your use case is to print the full description when entering a room, could use the `onEnter` hook? Like so:
```js
{
    id: 'roomid',
    name: 'Room Name',
    desc: `Long description of the room`,
    onEnter: () => {
        const room = getRoom('roomid')
        if(room.visits > 1) {
            println(room.desc)
        }
    }
}
```
On the first visit, the description will be printed by the engine. On successive visits, the description will be printed by this `onEnter` hook.

In order to apply it to each room, you would have to copy this to each room's `onEnter` or send the disk/rooms through a function to apply it programmatically.

**Comment by okaybenji** [comment, marked as answer]
Hi, thank you for your question and very glad you like the engine!

There are a number of ways to tackle this. @dthigpen's recommendation above is one way. The upside to this approach is that it doesn't change anything about how the engine works, which is a less brittle approach (because if the engine changed, your adaption might break). The downside is that it isn't very DRY (as in "don't repeat yourself"). *I should note here I'm referring to the idea of updating each room manually to do this. The suggestion of applying this change programmatically would certainly DRY it up.*

Another approach which would be more DRY but also more brittle would be to override the `enterRoom` command. You would only have to do this in one place rather than for every room. You might first capture how `enterRoom` is working in another variable, then override it to add this functionality like so:

```js
const defaultEnterRoom = enterRoom;
enterRoom = (id) => {
  defaultEnterRoom(id);
  const room = getRoom(id);
  if(room.visits > 1) {
    println(room.desc)
  }
};
```

Another way which is more brittle but also dead simple would be to simply change the `enterRoom` method in the engine itself. You would just comment out lines 972 and 974, disabling the `if` statement around the line that prints the description:

```js
 // if (room.visits === 0) {
    println(room.desc);
 // }
```
